pkg/infra/remotecache/redis_storage.go

```diff
@@ -22,9 +22,14 @@ func parseRedisConnStr(connStr string) (*redis.Options, error) {
 	keyValueCSV := strings.Split(connStr, ",")
 	options := &redis.Options{Network: "tcp"}
 	for _, rawKeyValue := range keyValueCSV {
-		keyValueTuple := strings.Split(rawKeyValue, "=")
+		keyValueTuple := strings.SplitN(rawKeyValue, "=", 2)
 		if len(keyValueTuple) != 2 {
-			return nil, fmt.Errorf("incorrect redis connection string format detected for '%v', format is key=value,key=value", rawKeyValue)
+			if strings.HasPrefix(rawKeyValue, "password") {
+				// don't log the password
+				return nil, fmt.Errorf("incorrect redis connection string format detected for 'password', format is key=value,key=value")
```

Comment:
Instead of returning a different error I would proposed to censor the `rawKeyValue`.
In that case you could do something like this:

```
	for _, rawKeyValue := range keyValueCSV {
		keyValueTuple := strings.SplitN(rawKeyValue, "=", 2)
        censored := []byte(rawKeyValue)
		if len(keyValueTuple) != 2 {
			if strings.HasPrefix(rawKeyValue, "password") {
                for i := len("password"); i < len(rawKeyValue); i++ {
                    censored[i] = '*'
                }
			}
            return nil, fmt.Errorf("incorrect redis connection string format detected for '%v', format is key=value,key=value", string(censored))
		}
```

Reply:
Instead of leaking any information (eg. length of the password) what about the current solution? I think nobody is interested in the correct length so the error should be enough.